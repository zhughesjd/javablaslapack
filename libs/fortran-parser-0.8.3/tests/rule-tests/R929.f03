! Test inquire-stmt
!      inquire-stmt  is  INQUIRE ( inquire-spec-list )
!                    or  INQUIRE ( IOLENGTH = scalar-int-variable ) 
!                          output-item-list
!
! Not tested here: inquire-spec-list, scalar-int-variable, and
!  output-item-list.
inquire(10)
inquire(10, file='foo.txt', blank=my_blank_variable)
inquire(iolength=x) a, b, c

end
