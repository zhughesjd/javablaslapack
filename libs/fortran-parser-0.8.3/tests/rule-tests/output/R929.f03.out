*******************************************
args[1]: R929.f03
R1101:main-program__begin:
R 204:specification-part: numUseStmts=0 numImportStmts=0 numDeclConstructs=0
R 930:inquire-spec-list__begin:
R 406:int-literal-constant: digitString=[@16,282:283='10',<10>,8:8] kindParam=null
R 306:literal-constant:
R 701:primary:
R 930:inquire-spec: specifier=null
R 930:inquire-spec-list: count=1
R 929:inquire-stmt: isType2=false
R 214:action-stmt:
R 213:executable-construct:
R 930:inquire-spec-list__begin:
R 406:int-literal-constant: digitString=[@21,294:295='10',<10>,9:8] kindParam=null
R 306:literal-constant:
R 701:primary:
R 930:inquire-spec: specifier=null
R 427:char-literal-constant: digitString-kind-param=null identifier-kind-param=null str=[@26,303:311=''foo.txt'',<8>,9:17]
R 306:literal-constant:
R 701:primary:
R 930:inquire-spec: specifier=[@24,298:301='file',<209>,9:12]
R 613:part-ref: id=[@31,320:336='my_blank_variable',<209>,9:34] hasSelectionSubscriptList=false hasImageSelector=false
R 612:data-ref: numPartRef=1
R -3:designator-or-func-ref:
R 701:primary:
R 930:inquire-spec: specifier=[@29,314:318='blank',<209>,9:28]
R 930:inquire-spec-list: count=3
R 929:inquire-stmt: isType2=false
R 214:action-stmt:
R 213:executable-construct:
R 209:execution-part-construct:
R 613:part-ref: id=[@38,356:356='x',<209>,10:17] hasSelectionSubscriptList=false hasImageSelector=false
R 612:data-ref: numPartRef=1
R 603:designator: hasSubstringRange=false
R 601:variable:
R 472:scalar-int-variable:
R 916:output-item-list__begin:
R 613:part-ref: id=[@41,359:359='a',<209>,10:20] hasSelectionSubscriptList=false hasImageSelector=false
R 612:data-ref: numPartRef=1
R -3:designator-or-func-ref:
R 701:primary:
R 916:output-item:
R 613:part-ref: id=[@44,362:362='b',<209>,10:23] hasSelectionSubscriptList=false hasImageSelector=false
R 612:data-ref: numPartRef=1
R -3:designator-or-func-ref:
R 701:primary:
R 916:output-item:
R 613:part-ref: id=[@47,365:365='c',<209>,10:26] hasSelectionSubscriptList=false hasImageSelector=false
R 612:data-ref: numPartRef=1
R -3:designator-or-func-ref:
R 701:primary:
R 916:output-item:
R 916:output-item-list: count=3
R 929:inquire-stmt: isType2=true
R 214:action-stmt:
R 213:executable-construct:
R 209:execution-part-construct:
R 208:execution-part:
R1103:end-program-stmt: id=null
R1101:main-program: hasProgramStmt=false hasExecutionPart=true hasInternalSubprogramPart=false
Parser exiting normally
********************************************
